### PR TITLE
CP-14037: Remove xenprep

### PR DESCRIPTION
**Pull request by thomassa**

### CODE REVIEW

File: ocaml/xapi/xapi_vm.ml [hunk 1]
```diff
@@ -1061,7 +1061,10 @@ let import ~__context ~url ~sr ~full_restore ~force =
 let query_services ~__context ~self =
 	raise (Api_errors.Server_error(Api_errors.not_implemented, [ "query_services" ]))
 
+(* Disabled for the time being, pending a decision on what form the feature will take *)
```
Comment (simonjbeaumont) [inline ocaml/xapi/xapi_vm.ml]: Let's use version control properly. Commenting out code is horrible!

**Comment by simonjbeaumont** [comment]
Also, are these changes (reverts) not achievable by reverting commits. That would give a better confidence that we're not gonna leave remnants in the codebase. Perhaps the nature of the original patch series doesn't make this possible. As we can see, not doing this is error prone as the build is failing.

File: ocaml/xapi/xapi_vm.ml [hunk 2]
```diff
@@ -1061,7 +1061,10 @@ let import ~__context ~url ~sr ~full_restore ~force =
...
 let assert_can_set_auto_update_drivers ~__context ~self ~value =
+	raise (Api_errors.Server_error(Api_errors.not_implemented, [ "set_auto_update_drivers" ]))
```
Comment (simonjbeaumont) [inline ocaml/xapi/xapi_vm.ml]: You can't do this, it leads to the unsound type error. You _can_ fool the type checker by putting:

```
if true then raise exn
```

but this really bad.

Is this API call in a released version? If not, can't we just nuke it?

**Comment by simonjbeaumont** [comment]
From xen-git:
```
15:44:26 File "xapi_vm.ml", line 1079, characters 1-59:
15:44:26 Warning 21: this statement never returns (or has an unsound type.)
15:44:27 File "xapi_vm.ml", line 1:
15:44:27 Error: Some fatal warnings were triggered (1 occurrences)
```

**Comment by thomassa** [comment]
For removing xenprep: no, I don't want to do that by reverting commits because there are some changes worth keeping.

For disabling set_auto_update_drivers... on second thoughts, I'll leave that for now. Might do it in a separate pull-request, or might just wait until we know what the feature is meant to do.

**Comment by johnelse** [comment]
Looks good to me.

**Pull request merged**